### ISSUE TITLE
Warn that DNS routing through Pi-hole (on Umbrel) blocks upgrade to v0.5

### ISSUE THREAD

**Issue by mhadam** [getumbrel/umbrel#1443]
Like the title says, this will prevent the v0.5 upgrade since the Docker package repository won't resolve. It's not exactly obvious this is happening unless you read through the logs.

**Comment by nevets963** [comment]
@mhadam Thanks for opening the issue! We're aware of the issue and I have a PR ready to address this :) We expect the fix to be available within the next Umbrel release.

**Comment by GalvaniObst** [comment]
I had the same issue.

So in case you get this in your logs:
```
+ sleep 1
+ sh -c apt-get update -qq >/dev/null
W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian-security/dists/buster/updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch http://deb.debian.org/debian/dists/buster-updates/InRelease  Temporary failure resolving 'deb.debian.org'
W: Failed to fetch https://download.docker.com/linux/debian/dists/buster/InRelease  Temporary failure resolving 'download.docker.com'
W: Failed to fetch http://archive.raspberrypi.org/debian/dists/buster/InRelease  Temporary failure resolving 'archive.raspberrypi.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
+ sh -c DEBIAN_FRONTEND=noninteractive apt-get install -y -qq apt-transport-https ca-certificates curl >/dev/null
+ sh -c mkdir -p /etc/apt/keyrings && chmod -R 0755 /etc/apt/keyrings
+ sh -c curl -fsSL "https://download.docker.com/linux/debian/gpg" | gpg --dearmor --yes -o /etc/apt/keyrings/docker.gpg
curl: (6) Could not resolve host: download.docker.com
gpg: no valid OpenPGP data found.
Updating Docker failed, reverting update!
sending incremental file list
```


edit /etc/resolv.conf and add a second DNS server, for example:
`nameserver 1.1.1.1`

After the update the second nameserver can be removed.

**Comment by nevets963** [comment]
We resolved this issue in Umbrel v0.5.1 that rolled out yesterday where we use public DNS servers during update process, and then on completion, return the old DNS config.
